src/utils/simplifier.js

```diff
@@ -103,14 +103,53 @@ function simplify(realm, value: Value, isCondition: boolean = false): Value {
         // (x: boolean) || false <=> x
         if (!y.mightNotBeFalse()) return op === "||" ? x : realm.intrinsics.false;
       }
-      if (
-        op === "||" &&
-        y instanceof AbstractValue &&
-        y.kind === "||" &&
-        x.equals(y.args[0]) &&
-        !y.args[1].mightNotBeTrue()
-      )
-        return y;
+      if (op === "||") {
+        if (y instanceof AbstractValue && y.kind === "||") {
+          // x || x || y <=> x || y
+          if (x.equals(y.args[0])) return y;
+          // y || x || y <=> x || y
+          if (x instanceof AbstractValue && x.kind === "!") {
+            // !x || x || y <=> y
```

Comment:
This (and the next) rule seems wrong, unless I am missing some special JavaScript craziness.

Reply:
The comment is correct if you don't equate the meta variable x with the comment variable x. I'll update the comment to make it more difficult to equate them.